### PR TITLE
feat(grid): add styled-system props (FE-4912)

### PR DESCRIPTION
**Pull request by grabkowski**
### Proposed behaviour
Add styled-system grid props to GridContainer and GridItem

### Current behaviour
There is no possibility to customize the grid layout using the Carbon Grid component

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [x] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

### CODE REVIEW

File: src/components/grid/grid-item/grid-item.component.js [hunk 1]
```diff
@@ -23,6 +23,8 @@ GridItem.propTypes = {
   children: PropTypes.node,
   /** How the grid item is aligned along the block (column) axis. Values: start, end, center, stretch */
   alignSelf: PropTypes.string,
+  /** Shorthand property for gridColumn and gridRow */
+  gridArea: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
```
Comment (samtjo) [inline src/components/grid/grid-item/grid-item.component.js:27]: We probably want to add this to the `responsiveSettings` prop types below as well I think

Comment (edleeks87) [reply src/components/grid/grid-item/grid-item.component.js:27]: I agree with Sam, I also had a little play around in a sandbox to refresh myself of how this component works and noticed the `gridArea` prop doesn't behave as I expect, the values do not override etc. 

I've set the same values on the element in the console to show what I'd expect to see what is currently seen in the styles

image


https://codesandbox.io/s/carbon-quickstart-forked-3xrlj8?file=/src/index.js

Comment (grabkowski) [reply src/components/grid/grid-item/grid-item.component.js:27]: that was caused by default values being set for `grid-column` and `grid-row`. I've replaced these defaults with `grid-area` now.

File: src/components/grid/grid-item/grid-item.d.ts [hunk 2]
```diff
@@ -21,10 +26,12 @@ export interface GridItemProps extends PaddingProps {
   justifySelf?: string;
   /** Defines the Component(s) to be rendered within the GridItem */
   children?: React.ReactNode;
+  /** Shorthand property for gridColumn and gridRow */
+  gridArea?: GridAreaProps;
```
[the remark this answers is not in the document]
Comment (grabkowski) [reply src/components/grid/grid-item/grid-item.d.ts:30]: added

**Review by edleeks87 — APPROVED** [review]

**Review by samtjo — APPROVED** [review]

**Review by samtjo — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by samtjo — APPROVED** [review]

**Review by edleeks87 — DISMISSED** [review]

**Review by edleeks87 — DISMISSED** [review]

**Review by samtjo — DISMISSED** [review]

**Review by DlgSHi — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by samtjo — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**